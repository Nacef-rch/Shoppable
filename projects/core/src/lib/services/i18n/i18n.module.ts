import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { TranslateModule, TranslateLoader } from '@ngx-translate/core';
import { TranslateHttpLoader } from '@ngx-translate/http-loader';
import { HttpClientModule, HttpClient } from '@angular/common/http';
import { I18ntestComponent } from './components/i18ntest/i18ntest.component';
import { I18nService } from './services/i18n.service';
export function HttpLoaderFactory(http: HttpClient) {
    return new TranslateHttpLoader(http);
}

@NgModule({
    declarations: [I18ntestComponent],
    imports: [
        CommonModule,
        HttpClientModule,
        TranslateModule.forRoot({
            loader: {
                provide: TranslateLoader,
                useFactory: HttpLoaderFactory,
                deps: [HttpClient]
            }
        })
    ],
    providers: [I18nService],
    exports: [I18ntestComponent, TranslateModule]
})
export class I18nModule {}
