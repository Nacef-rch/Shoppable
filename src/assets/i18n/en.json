{
    "AUTH": {
        "LOGIN": {
            "CREATE": "Create an account",
            "SIGN": "Sign In",
            "VALID_EMAIL": "Please enter a valid email !",
            "EMAIL": "Email",
            "PASS_IS": "Password is",
            "EMAIL_IS": "Email is",
            "REQUIRED": "required",
            "PASS": "Enter your password",
            "REMEMBER": "Remember me",
            "LOG_IN": "Log in",
            "LOG_WITH": "Or login with"
        },
        "REGISTER": {
            "SIGN": "Sign up",
            "MEMBER": "I am already member",
            "VALID_NAME": "Please enter your name !",
            "NAME": "Name",
            "VALID_EMAIL": "Please enter a valid email !",
            "EMAIL": "Email",
            "VALID_PASS": "Please enter a password !",
            "PASS": "Password",
            "VALID_REPASS": "Please repeat your password !",
            "REPASS": "Repeat your password",
            "VALID_HANDLE": "Please enter a Handle !",
            "HANDLE": "Handle",
            "AGREE": "I agree all statements in",
            "TERMS": "Terms of service",
            "REGISTER": "Register",
            "NAME_IS": "Name is ",
            "EMAIL_IS": "Email is",
            "PASS_IS": "Password is",
            "HANDLE_IS": "Handle is"
        }
    },
    "DASH": {
        "FOOTER": {
            "COPY_RIGHTS": "Copyright 2020",
            "HOME": "Store"
        },
        "GET_PRODUCTS": {
            "PRODUCT": "Product",
            "PRODUCTS": "Products",
            "CATEGORY": "Category",
            "ADD": "Added by",
            "INCOMING": "Incoming",
            "AVAILABLE": "Available",
            "EDIT": "Edit quantity available",
            "SAVE": "Save"
        },
        "USER_BOX": {
            "SETT": "Settings",
            "LOG": "Logout"
        },
        "SIDE_BAR": {
            "PRODUCTS": "Products",
            "ALL": "All Products",
            "ADD": "Add your products",
            "INVENTORY": "Inventory",
            "ORDERS": "Purchase orders",
            "GIFT": "Gift Card",
            "ANALYTICS": "Analytics",
            "DASHBOARDS": "Dashboard",
            "REPORTS": "Reports"
        },
        "INVENTORY": {
            "CARD": "Card view",
            "SELECT": "select a product",
            "HEAD": "Inventory",
            "SUB": "In this page you can organize all of your products"
        },
        "IMPORT_PRODUCTS": {
            "PRODUCT": "Product",
            "TITLE": "Title",
            "FIELD": "This field",
            "TYPE": "Product type",
            "SELECT": " Select a",
            "CATEGORY": "category !",
            "SELECTED_CATEGORY": "Selected category :",
            "DESCRIPTION": "Description",
            "FILE": "File",
            "IMAGE": "The image must be in ",
            "OR": "or",
            "FORMAT": "format !",
            "PRICING": "Pricing",
            "QUANTITY": "Quantity",
            "PRICE": "Price",
            "AVAILABLE": "Available",
            "SUBMIT": "Submit",
            "CARD": "Card preview",
            "HEAD": "Add product",
            "SUB": "In this page you can add your products"
        },
        "WELCOME": {
            "YOUR": "Add your products",
            "GET": "Get closer to your first sale by adding products, or import your existing product inventory.",
            "ADD": "Add product"
        },
        "SEARCH": { "SEARCH": "Type to search" },
        "LIST": {
            "INV": "Inventory",
            "STAT": "Statistics",
            "SET": "Settings"
        }
    }
}
