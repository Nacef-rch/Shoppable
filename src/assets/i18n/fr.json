{
    "AUTH": {
        "LOGIN": {
            "CREATE": "Créer un compte",
            "SIGN": "Se connecter",
            "VALID_EMAIL": "Veuillez saisir un Email valide !",
            "EMAIL": "Votre Email",
            "PASS_IS": "Mot de passe",
            "EMAIL_IS": "Email est",
            "REQUIRED": "requis",
            "PASS": "Mot de passe",
            "REMEMBER": "Rester connecté",
            "LOG_IN": "S'identifier",
            "LOG_WITH": "Ou connectez-vous avec"
        },
        "REGISTER": {
            "SIGN": "S'inscrire",
            "MEMBER": "J'ai un compte",
            "VALID_NAME": "Entrez votre nom ici !",
            "NAME": "Votre nom",
            "VALID_EMAIL": "Veuillez saisir un email valide!",
            "EMAIL": "Votre Email",
            "VALID_PASS": "Veuillez entrer un mot de passe !",
            "PASS": "Mot de passe",
            "VALID_REPASS": "Veuillez répétez votre mot de passe!",
            "REPASS": "Répétez votre mot de passe",
            "VALID_HANDLE": "Veuillez entrer un Handle !",
            "HANDLE": "Handle",
            "AGREE": "J'accepte toutes les déclarations des",
            "TERMS": "conditions d'utilisation",
            "REGISTER": "S'inscrire",
            "NAME_IS": "Nom est ",
            "EMAIL_IS": "Email est",
            "PASS_IS": "Mot de passe",
            "HANDLE_IS": "Handle est"
        }
    },
    "DASH": {
        "FOOTER": {
            "COPY_RIGHTS": "Copyright 2020",
            "HOME": "Boutique"
        },
        "GET_PRODUCTS": {
            "PRODUCT": "Produit",
            "PRODUCTS": "Produits",
            "CATEGORY": "Catégorie",
            "ADD": "Ajouté par",
            "INCOMING": "revenu",
            "AVAILABLE": "Disponible",
            "EDIT": "Modifier la quantité",
            "SAVE": "Save"
        },
        "USER_BOX": {
            "SETT": "Réglages",
            "LOG": "Se déconnecter"
        },
        "SIDE_BAR": {
            "PRODUCTS": "Produits",
            "ALL": "Tous les produits",
            "ADD": "Ajoutez vos produits",
            "INVENTORY": "Inventaire",
            "ORDERS": "Les achat en ligne",
            "GIFT": "Carte cadeau",
            "ANALYTICS": "Analytique",
            "DASHBOARDS": "Dashboard",
            "REPORTS": "Rapports"
        },
        "INVENTORY": {
            "CARD": "Aperçu de la carte",
            "SELECT": "choisissez un produit"
        },
        "IMPORT_PRODUCTS": {
            "PRODUCT": "Produit",
            "TITLE": "Titre",
            "FIELD": "Ce champ",
            "TYPE": "Type de produit",
            "SELECT": " Sélectionner une",
            "CATEGORY": "catégorie !",
            "SELECTED_CATEGORY": "Catégorie sélectionnée:",
            "DESCRIPTION": "Description",
            "FILE": "Fichier",
            "IMAGE": "L'image doit être au format ",
            "OR": "ou",
            "FORMAT": "",
            "PRICING": "Tarification",
            "QUANTITY": "Quantité",
            "PRICE": "Prix",
            "AVAILABLE": "Disponible",
            "SUBMIT": "Envoyé",
            "CARD": "Aperçu de la carte"
        }
    }
}
