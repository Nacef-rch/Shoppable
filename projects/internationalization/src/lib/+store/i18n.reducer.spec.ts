import * as I18nActions from '@i18n/+store/i18n.actions';
import {
    i18nInitialState as initialState,
    i18nReducer as reducer
} from '@i18n/+store/i18n.reducer';

describe('I18nReducer', () => {
    describe('undefined action', () => {
        it('should return the default state', () => {
            // GIVEN
            // WHEN
            const result = reducer(undefined, {} as any);
            // THEN
            expect(result).toMatchSnapshot();
        });
    });

    describe('SET_LANGUAGE', () => {
        it('should set lang to `fr` in state', () => {
            // GIVEN
            const lang = 'fr';
            const action = I18nActions.setLanguage({ lang });
            // WHEN
            const result = reducer(initialState, action);
            // THEN
            expect(result).toMatchSnapshot();
        });
    });
});
