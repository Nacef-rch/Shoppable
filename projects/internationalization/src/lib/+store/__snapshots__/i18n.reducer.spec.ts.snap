// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`I18nReducer SET_LANGUAGE should set lang to \`fr\` in state 1`] = `
Object {
  "lang": "fr",
}
`;

exports[`I18nReducer undefined action should return the default state 1`] = `
Object {
  "lang": "fr",
}
`;
